### PR TITLE
use int32 in tests where possible

### PR DESCRIPTION
**Pull request by andredaprato**
I don't like changing int8 to int32 but we won't be able to run these tests on webgpu without changing it. The rest of the changes avoid implicitly creating long tensors due to numpy/torch initialization which we do in most places already.

### CODE REVIEW

**Comment by geohot** [comment]
webgpu doesn't support int8?

**Comment by andredaprato** [comment]
No, only 32 bit scalars and f16 are supported in the spec, https://www.w3.org/TR/WGSL/#host-shareable-types.

File: test/test_ops.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def compare(s, tinygrad_output, torch_output, atol, rtol):
 
 def prepare_test_op(low, high, shps, vals, forward_only=False):
   if shps is None:
-    ts = [torch.tensor(x, requires_grad=(not forward_only)) for x in vals]
+    ts = [(t:=torch.tensor(x, requires_grad=(not forward_only))).to(torch.int32 if not torch.is_floating_point(t) else None) for x in vals]
```
Comment (geohot) [inline test/test_ops.py:69]: I really don't like this change, I can't predict what it does

File: test/test_uops_stats.py [hunk 2]
```diff
@@ -17,48 +17,48 @@ def get_stats(x:Tensor):
 
 class TestMemoryCount(unittest.TestCase):
```
Comment (geohot) [inline test/test_uops_stats.py:18]: This should be a unit test anyway

**Comment by geohot** [comment]
I don't like PRs like this. This is not an improvement to tinygrad, this is a weird prereq that's changing tests.

How can you do this stuff while improving tinygrad? If a test doesn't test anything accelerator specific, it should be in test/unit then you don't have to worry about it.

File: test/test_rearrange_einops.py [hunk 3]
```diff
@@ -164,7 +164,7 @@ def test_rearrange_ellipsis_ops(self):
       ("a b c d e -> b (a c d) e", "a b ... e -> b (a ...) e"),
     ]
 
-    xnp = np.arange(2 * 3 * 4 * 5 * 6).reshape([2, 3, 4, 5, 6])
+    xnp = np.arange(2 * 3 * 4 * 5 * 6, dtype=np.int32).reshape([2, 3, 4, 5, 6])
```
Comment (geohot) [inline test/test_rearrange_einops.py:167]: This test should be refactored, not have uint32 spammed everywhere

**Pull request closed without merging**